source/data-modeling/associations.txt

```diff
@@ -0,0 +1,504 @@
+   :language: ruby
+   :start-after: # start-has-many-any
+   :end-before: # end-has-many-any
+
+You can also use the ``any?`` method with a filter to find documents that match
+a specified criteria, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-many-any-filter
+   :end-before: # end-has-many-any-filter
+
+You can supply a class name to the ``any?`` method to filter the results by the
+name of the class. This is useful for polymorphic associations:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-many-any-class
+   :end-before: # end-has-many-any-class
+
+.. note::
+
+   After the data of the associated class is loaded to {+odm+}, subsequent calls
+   to the ``any?`` method do not query the database. Instead, {+odm+} uses the
+   data that is already loaded in memory.
+
+You can also call the ``exists?`` method to determine if there are any persisted
+documents in the association. The ``exists?`` method always queries the
+database, and checks only for documents that have been saved to the database.
+The ``exists?`` method does not allow for filtering and does not accept any
+arguments.
+
+The following example uses the ``exists?`` method to determine if there are any persisted
+``Members`` documents in the ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-has-many-exists
+   :end-before: # end-has-many-exists
+
+.. _mongoid-belongs-to:
+
+Belongs To
+~~~~~~~~~~
+
+Use the ``belongs_to`` macro to declare that a class model is a child of another
+class. By default, the ``_id`` field of the parent class is stored in the child
+class. The following example creates a ``Members`` class with a ``belongs_to``
+association to a ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-belongs-to
+   :end-before: # end-belongs-to
+
+You can allow {+odm+} to persist documents to the database without storing the
+``_id`` of the associated parent class by setting the ``optional`` option to
+``true``, as shown in the following example:
+
+.. literalinclude:: /includes/data-modeling/associations.rb
+   :language: ruby
+   :start-after: # start-belongs-to-optional
+   :end-before: # end-belongs-to-optional
+
+.. tip::
+
+   You can globally change the default behavior of the ``belongs_to``
+   association to not require their parent class by setting the
+   ``belongs_to_required_by_default`` configuration option to ``false`` in your
+   application's configuration settings.
+
+You can specify a ``belongs_to`` in a child class without specifying a matching
+``has_one`` or ``has_many`` association in the parent class. When doing so, the
+documents of the parent class are accessible from the child class, but the child
```

Comment:
Q: what does it mean for the parent models to be accessible from the child? A document in Members has a field whose value is a document in the Band class?

Reply:
Reworded this. The child class automatically stores the _id field of the parent, so that is accessible in this case.